notebooks/1.data_splits/nbconverted/1.data_splits.py

```diff
@@ -507,17 +631,17 @@ def get_and_rename_injury_info(
 
 # Data splits train test summary
 injury_train_info_df = get_and_rename_injury_info(
-    profile=X_train.merge(
-        fs_profile_df[meta_cols], how="left", right_index=True, left_index=True
-    )[meta_cols + fs_feats],
+    profile=aligned_X_train.merge(
+        aligned_fs_profile_df[meta_cols], how="left", right_index=True, left_index=True
```

Comment:
it's important to confirm that the indexes don't update somewhere, otherwise this will have unintended consequences with metadata not being associated with the correct features

Reply:
That’s a great point! I’ve added a check for the index merge. It works by comparing the raw data to verify if any values have changed since the start. This is done using the pandas.equals() function to ensure that the merged data and the original raw data are identical and contain the same values.